scopesim/commands/user_commands.py

```diff
@@ -172,113 +244,143 @@ def update(self, **kwargs):
-                        defyam["properties"]["modes"].append(mode)
-                        if "deprecate" in self.modes_dict[mode]:
-                            logger.warning(self.modes_dict[mode]["deprecate"])
-                    else:
-                        raise ValueError(f"mode '{mode}' was not recognised")
-
-        self.__init__(yamls=self.default_yamls)
-
-    def list_modes(self):
-        if isinstance(self.modes_dict, Mapping):
-            modes = {}
-            for mode_name in self.modes_dict:
-                dic = self.modes_dict[mode_name]
-                desc = dic["description"] if "description" in dic else "<None>"
-                modes[mode_name] = desc
-                if "deprecate" in dic:
-                    modes[mode_name] += " (deprecated)"
-
-            msg = "\n".join([f"{key}: {value}" for key, value in modes.items()])
+    @staticmethod
+    def update_alias(mapping: MutableMapping, new_dict: Mapping) -> None:
+        """Update a dict-like according to the alias-properties syntax.
+
+        This used to be part of `astar_utils.NestedMapping`, but is specific to
+        ScopeSim and thus belongs somewhere here. It should only be used in the
+        context of YAML-dicts loaded by UserCommands, hence it was put here.
+        """
+        if isinstance(new_dict, NestedMapping):
+            new_dict = new_dict.dic  # Avoid updating with another one
+
+        if alias := new_dict.get("alias"):
+            logger.debug("updating alias %s", alias)
+            propdict = new_dict.get("properties", {})
+            if alias in mapping:
+                mapping[alias] = recursive_update(mapping[alias], propdict)
+            else:
+                mapping[alias] = propdict
         else:
-            msg = "No modes found"
-        return msg
+            # Catch any bang-string properties keys
+            to_pop = []
+            for key in new_dict:
+                if is_bangkey(key):
+                    logger.debug(
+                        "Bang-string key %s was seen in .update. This should "
+                        "not occur outside mocking in testing!", key)
+                    mapping[key] = new_dict[key]
+                    to_pop.append(key)
+            for key in to_pop:
+                new_dict.pop(key)
+
+            if len(new_dict) > 0:
+                mapping = recursive_update(mapping, new_dict)
 
-    @property
-    def modes(self):
-        print(self.list_modes())
 
-    def __setitem__(self, key, value):
-        self.cmds.__setitem__(key, value)
+    def set_modes(self, *modes) -> None:
+        """Reload with the specified `modes`.
 
-    def __getitem__(self, item):
-        return self.cmds.__getitem__(item)
+        .. versionchanged:: PLACEHOLDER_NEXT_STABLE_VERSION
 
-    def __contains__(self, item):
-        return self.cmds.__contains__(item)
+        This used to take a single list-like argument, now used a "*args"
+        approach to deal with multiple modes.
```

Comment:
There will be scripts and notebooks lying around on peoples drives that will pass a list to `set_modes()`. I strongly believe that we should keep supporting those scripts. It seems perfectly possible to make the changes in this PR while preserving backwards compatibility.

Could you please see if it is possible to write `set_modes()` so it can still also accept a list?

Reply:
Oh for some reason I thought that `set_modes()` was also called in a notebook or so. I didn't pay enough attention. If it is only in this one test, then well, it would probably be fine to change it.

Reply:
I ended up going with the DeprecationWarning approach in [d31c490](https://github.com/AstarVienna/ScopeSim/pull/375/commits/d31c49029fe8833bf7b60a43fa250bdd4ffcc600).